packages/cardano-services/src/Projection/createTypeormProjection.ts

```diff
@@ -118,15 +118,23 @@ export const createTypeormProjection = ({
     defer(() =>
       projectionSource$.pipe(
         applyMappers(mappers),
-        shareRetryBackoff(
-          (evt$) =>
-            evt$.pipe(
-              withTypeormTransaction({ connection$: connect() }),
-              applyStores(stores),
-              buffer.storeBlockData(),
-              typeormTransactionCommit()
-            ),
-          { shouldRetry: isRecoverableTypeormError }
+        // if there are any relevant data to write into db
```

Comment:
Please add a test/assertion that find an empty block in the dataset and asserts that blocks table doesn't contain it

Reply:
Added the test here https://github.com/input-output-hk/cardano-js-sdk/pull/1200/commits/4bfa2c0db8a8f7ac8d777c77478f0161f567c1a6 , but implemented it a bit different, I created a test file with a few blocks and only one with mint, and then asserted only one block was stored

Reply:
updated here https://github.com/input-output-hk/cardano-js-sdk/pull/1200/commits/3c09a5575b5348845768ecb56bea8b495f3327a5